jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/Unit.java

```diff
@@ -0,0 +1,152 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.deploy.providers;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.eclipse.jetty.deploy.App;
+
+/**
+ * A Unit of deployment, a basename and all the associated
+ * paths that belong to that basename, along with state.
+ */
+public class Unit
```

Comment:
I think that `Unit` should be promoted to a top level class, and `App` demoted to an implementation detail. The `Unit` class better captures what we're dealing with - a possible collection of files and/or directories that taken all together represent a deployable.  The `App` is really just a subset of that information.

Reply:
Can you clarify succinctly the difference between `App` and `Unit` using language like "A Unit is a ....", "An App is a ..."? Looking at the code `App` has `Path` references and so does `Unit`, so in that sense they both refer to resources that can be disk based resources, so that can't be the difference.